src/queries/role.js

```diff
@@ -1,48 +1,64 @@
-import {getImplicitAriaRoles, prettyRoles} from '../role-helpers'
+import {
+  getImplicitAriaRoles,
+  prettyRoles,
+  shouldExcludeFromA11yTree,
+} from '../role-helpers'
 import {buildQueries, fuzzyMatches, makeNormalizer, matches} from './all-utils'
 
 function queryAllByRole(
   container,
   role,
-  {exact = true, collapseWhitespace, trim, normalizer} = {},
+  {exact = true, collapseWhitespace, hidden = false, trim, normalizer} = {},
 ) {
   const matcher = exact ? matches : fuzzyMatches
   const matchNormalizer = makeNormalizer({collapseWhitespace, trim, normalizer})
 
-  return Array.from(container.querySelectorAll('*')).filter(node => {
-    const isRoleSpecifiedExplicitly = node.hasAttribute('role')
+  return Array.from(container.querySelectorAll('*'))
+    .filter(element => {
+      return hidden === false
+        ? shouldExcludeFromA11yTree(element) === false
+        : true
+    })
+    .filter(node => {
+      const isRoleSpecifiedExplicitly = node.hasAttribute('role')
 
-    if (isRoleSpecifiedExplicitly) {
-      return matcher(node.getAttribute('role'), node, role, matchNormalizer)
-    }
+      if (isRoleSpecifiedExplicitly) {
+        return matcher(node.getAttribute('role'), node, role, matchNormalizer)
+      }
 
-    const implicitRoles = getImplicitAriaRoles(node)
+      const implicitRoles = getImplicitAriaRoles(node)
 
-    return implicitRoles.some(implicitRole =>
-      matcher(implicitRole, node, role, matchNormalizer),
-    )
-  })
+      return implicitRoles.some(implicitRole =>
+        matcher(implicitRole, node, role, matchNormalizer),
+      )
+    })
 }
 
 const getMultipleError = (c, role) =>
   `Found multiple elements with the role "${role}"`
 
-const getMissingError = (container, role) => {
-  const roles = prettyRoles(container)
+const getMissingError = (container, role, {hidden = false} = {}) => {
+  const roles = prettyRoles(container, {hidden})
   let roleMessage
 
   if (roles.length === 0) {
-    roleMessage = 'There are no available roles.'
+    if (hidden === false) {
+      roleMessage = 'There are no accessible roles.'
+    } else {
+      roleMessage = 'There are no available roles.'
+    }
   } else {
     roleMessage = `
-Here are the available roles:
+Here are the ${hidden === false ? 'accessible' : 'available'} roles:
 
   ${roles.replace(/\n/g, '\n  ').replace(/\n\s\s\n/g, '\n\n')}
 `.trim()
   }
 
   return `
-Unable to find an element with the role "${role}"
+Unable to find an ${
+    hidden === false ? 'accessible ' : ''
```

Comment:
Remove the extra space.

```suggestion
    hidden === false ? 'accessible' : ''
```

Reply:
I think I need to keep this and remove the unconditional space later on. Otherwise we get

```
Unable to find an  element with the role "${role}
                 ^^ double space
```
for `hidden: false`

Reply:
Extra space is still there (since I'm pretty sure it's required) but I fixed the double space issue.